saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncPersister.kt

```diff
@@ -81,8 +82,14 @@ class SavedSitesSyncPersister @Inject constructor(
     ): SyncMergeResult {
         val result = when (val validation = validateChanges(changes)) {
             is SyncDataValidationResult.Error -> SyncMergeResult.Error(reason = validation.reason)
-            is SyncDataValidationResult.Success -> processEntries(validation.data, conflictResolution)
-            else -> Success(false)
+            is SyncDataValidationResult.Success -> {
+                processEntries(validation.data, conflictResolution)
+            }
+
+            else -> {
+                updateSavedSitesMetadataWhenNoRemoteChanges()
```

Comment:
I don't think this is correct.

This is the case where BE responds with 304 (aka. Not modified).

There's one scenario I can think about where we can land here with local changes: hitting a rate limit.
- users will have local data but sync is paused because they hit a limit
- they are making changes, so we are trying to patch
- if BE accepts the changes, it will return 200 (e.g: changes fixed the limit error)
- if BE doesn't accept, it will return 304 because is same error response as in previous patches

if you get 304, that means changes not accepted, so it's not correct to copy request to reponse as `NoRemotechanges`.

Reply:
why would a 304 land in OnSuccess? If it does it’s wrong, features shouldn’t have to make a distinction based on a BE response code. let’s check tomorrow